include/maths/CLinearAlgebraEigen.h

```diff
@@ -271,8 +275,58 @@ class CDenseVector : public Eigen::Matrix<SCALAR, Eigen::Dynamic, 1> {
         }
         return seed;
     }
+
+    std::vector<SCALAR> toStdVector() const {
+        std::vector<SCALAR> result;
+        for (int i = 0; i < this->size(); ++i) {
+            result.push_back(this->operator()(i));
+        }
+        return result;
+    }
+
+    static CDenseVector<SCALAR> fromStdVector(const std::vector<SCALAR>& vector) {
+        CDenseVector<SCALAR> result(vector.size());
+        for (auto element : vector) {
+            result << element;
+        }
+        return result;
+    }
+
+    //! Persist by passing information to \p inserter.
+    static void persist(const CDenseVector<SCALAR>& vector,
+                        core::CStatePersistInserter& inserter) {
+        std::vector<SCALAR> temporaryVector;
+        for (int i = 0; i < vector.size(); ++i) {
+            temporaryVector.push_back(vector(i));
+        }
+        inserter.insertValue(DENSE_VECTOR_TAG, core::CPersistUtils::toString(temporaryVector));
+    }
+
+    //! Populate the object from serialized data
+    static bool restore(const std::string& tag,
+                        CDenseVector<SCALAR>& vector,
+                        core::CStateRestoreTraverser& traverser) {
```

Comment:
As above, if you implement this as non-static acceptRestoreTraverser this magically works with `core::CPersistUtils` also again this could use `fromStdVector`.

Reply:
The method was replaced by `acceptRestoreTraverser`, however I am not using `fromStdVector` on `this` to prevent memory leakage.

Reply:
It shouldn't leak memory. You'd need to do something like `*this = fromStdVector(vector)`. The assignment operator would do the clean up any memory held by `this`.

Reply:
I tried it, but I've got an Eigen asserting error during test execution. Therefore, I defaulted to the explicit copy.

Reply:
Hmm interesting. It says shouldn't do that unless the type has explicit sizes in the template declaration. This should be using dynamic so that ought not to be an issue.